### PR TITLE
[Android] Close some leaking resources.

### PR DESCRIPTION
**Pull request by rakuco**
Close a few InputStream and Cursor instances that were currently
dangling.

### CODE REVIEW

**Comment by rakuco** [comment]
ping @lincsoon @XiaosongWei

File: app/tools/android/test_data/extensions/contactextension/src/com/example/extension/ContactExtension.java [hunk 1]
```diff
@@ -88,16 +88,13 @@ String readContact(String name) {
...
-                String phone = phones.getString(phones.getColumnIndex(ContactsContract.CommonDataKinds.Phone.NUMBER));
-                phones.close();
-                cursor.close();
-                return phone;
-            } else {
-                Log.w(TAG, "no phone number for " + name);
-                cursor.close();
-                return "";
+            String phone = "";
+            if (phones.getCount() > 0) {
```
Comment (XiaosongWei) [inline app/tools/android/test_data/extensions/contactextension/src/com/example/extension/ContactExtension.java]: The cursor returned from getContentResolver().query may be null but no null check here.

**Comment by rakuco** [comment]
thanks, patch v2 is up

**Comment by XiaosongWei** [comment]
LGTM

File: app/tools/android/test_data/extensions/contactextension/src/com/example/extension/ContactExtension.java [hunk 2]
```diff
@@ -75,33 +75,29 @@ public void onMessage(int instanceID, String message){
...
+        try {
+            // Find the contacts by name.
+            nameCursor = cActivity.getContentResolver().query(ContactsContract.Contacts.CONTENT_URI,
+                    null, ContactsContract.Contacts.DISPLAY_NAME + "='" + name + "'", null, null);
+            if (nameCursor != null && nameCursor.getCount() >= 1) {
+                // Get the first phone of the specified name.
+                phoneCursor = cActivity.getContentResolver().query(
+                        ContactsContract.CommonDataKinds.Phone.CONTENT_URI, null,
+                        ContactsContract.Contacts.DISPLAY_NAME + "='" + name + "'", null, null);
+                if (phoneCursor != null && phoneCursor.getCount() >= 1) {
```
Comment (lincsoon) [inline app/tools/android/test_data/extensions/contactextension/src/com/example/extension/ContactExtension.java]: I think `if (phoneCursor != null && phoneCursor.moveToFirst())` is enough. `moveToFirst()` will return false if the cursor is empty

**Comment by rakuco** [comment]
patch v3 is up addressing Lin's comment

**Comment by lincsoon** [comment]
LGTM

**Pull request merged**